### ISSUE TITLE
All moderators should be able to upload presentations, not only the current presenter.

### ISSUE THREAD

**Issue by sthibaul** [bigbluebutton/bigbluebutton#10950]
Hello,

**Is your feature request related to a problem? Please describe.**
When we have several presentations made by different people one after the other in the same bbb room, each of them would like to be able to upload their .pdf slidedeck file, typically during the talk of the previous person. But only the presenter has the slidedeck loading item in the + menu.  That means that one has to wait for the end of the previous talk, switch presentation status, wait for the person to upload the file, wait for the conversion, etc. until eventually getting the slides shown, while all of this could be done during the previous presentation.

Of course, we want only the presenter to manipulate the currently shown slide deck etc. and change the currently shown slidedeck

**Describe the solution you'd like**

All moderators should be able to upload files independently of the presentation that is going on

**Describe alternatives you've considered**

Of course we can try to upload slides ahead of time, but in practice people bring their slides late...

**Comment by sthibaul** [comment]
We have again lost five minutes just waiting for the upload + conversion to happen...

**Comment by Subdarub** [comment]
@sthibaul I agree that the conversion process can take a lot of time, to circumvent this issue you can convert the file to a pdf before uploading it to BBB. This should save you a lot of time and may improve the quality of your presentations.

**Comment by sthibaul** [comment]
Yes, making it a pdf makes the conversion faster, but for presenters, it still takes significant time (especially when they are not used to bbb) to find out how to upload their slidedeck and actually upload it (which may be slow due to limited network), while this could be overlapped with the previous presentation.

**Comment by Entretoize** [comment]
I agree it's anoying that admins need to become a presenter to upload, maybe it's possible to automatically swith admin to presenter if he start an upload with a little trick ?

**Comment by sthibaul** [comment]
I don't want admins to become presenter, I believe we still want only one presenter at a time.
We just need to let admins upload files at any time.

**Comment by mark-franz** [comment]
I understand this idea, it's also relevant for my group of users. 
However I also experience that uploading a presentation can cause a decrease in audio and video quality for the conference as a whole. We typically wait until the presentation has been uploaded completely, until we continue talking. (Okay this also has to do with limited server resources, but upgrading this is also not so easy for us).
So I think it can be useful but we also should consider the implications and what it means for the actual use.
 
If, however, each presenter could pre-upload a presentation before or at the beginning of the meeting, that could resolve this, avoid lots of coordination between different presenters and the moderator ... 

Meanwhile, I strongly believe that a break between two presentations could be welcomed by participants as well :-)

**Comment by sthibaul** [comment]
> Okay this also has to do with limited server resources, but upgrading this is also not so easy for us

Ok, sure, but that does not mean that the feature should not be made available to those who have a strong-enough server :)

(in our case, we have many users, and most often several completely unrelated rooms open at the same time, so uploads can already happen at any time in other rooms anyway)

> Meanwhile, I strongly believe that a break between two presentations could be welcomed by participants as well :-)

Yes, a break is often welcome, but quite often the presenter realizes that the format is not supported, has to first convert to the another, support format, etc. the usual technical problems that do happen now and then, and shouldn't turn the 5-minute break into 15-minute shift in the whole day schedule.
